File: ticketing_systemFrontend/src/components/Configuration.js
import React from "react";

const ConfigurationForm = ({
  totalTickets,
  setTotalTickets,
  maxTicketCapacity,
  setMaxTicketCapacity,
  totalCustomers,
  setTotalCustomers,
  ticketReleaseRate,
  setTicketReleaseRate,
  customerRetrievalRate,
  setCustomerRetrievalRate,
  totalVendors,
  setTotalVendors,
}) => {
  // Helper function to validate and set the input value
  const handleInputChange = (setter) => (e) => {
    const value = e.target.value.trim();
    if (/^\d*$/.test(value)) {
      setter(value === "" ? "" : parseInt(value)); // Allow empty string or valid number
    } else {
      alert("Please enter a valid number.");
    }
  };

  return (
    <div className="form-container">
      <div className="form-group">
        <label>Total Tickets:</label>
        <input
          type="text"
          value={totalTickets}
          onChange={handleInputChange(setTotalTickets)}
        />
      </div>
      <div className="form-group">
        <label>Maximum Ticket Capacity:</label>
        <input
          type="text"
          value={maxTicketCapacity}
          onChange={handleInputChange(setMaxTicketCapacity)}
        />
      </div>
      <div className="form-group">
        <label>Total Customers:</label>
        <input
          type="text"
          value={totalCustomers}
          onChange={handleInputChange(setTotalCustomers)}
        />
      </div>
      <div className="form-group">
        <label>Ticket Release Rate (sec):</label>
        <input
          type="text"
          value={ticketReleaseRate}
          onChange={handleInputChange(setTicketReleaseRate)}
        />
      </div>
      <div className="form-group">
        <label>Customer Retrieval Rate (sec):</label>
        <input
          type="text"
          value={customerRetrievalRate}
          onChange={handleInputChange(setCustomerRetrievalRate)}
        />
      </div>
      <div className="form-group">
        <label>Number of Vendors:</label>
        <input
          type="text"
          value={totalVendors}
          onChange={handleInputChange(setTotalVendors)}
        />
      </div>
    </div>
  );
};

export default ConfigurationForm;

File: ticketing_systemBackend/src/main/java/com/example/ticketing_system/Configuration.java
package com.example.ticketing_system;

import java.io.Serializable;

public class Configuration implements Serializable {
    private int totalTickets;
    private int maxTicketCapacity;
    private int totalCustomers;
    private int ticketReleaseRate;
    private int customerRetrievalRate;
    private int totalVendors;

    // Constructor
    public Configuration(int totalTickets, int maxTicketCapacity, int totalCustomers,
                         int ticketReleaseRate, int customerRetrievalRate, int totalVendors) {
        this.totalTickets = totalTickets;
        this.maxTicketCapacity = maxTicketCapacity;
        this.totalCustomers = totalCustomers;
        this.ticketReleaseRate = ticketReleaseRate;
        this.customerRetrievalRate = customerRetrievalRate;
        this.totalVendors = totalVendors;
    }

    // Getters and setters
    public int getTotalTickets() { return totalTickets; }
    public void setTotalTickets(int totalTickets) { this.totalTickets = totalTickets; }

    public int getMaxTicketCapacity() { return maxTicketCapacity; }
    public void setMaxTicketCapacity(int maxTicketCapacity) { this.maxTicketCapacity = maxTicketCapacity; }

    public int getTotalCustomers() { return totalCustomers; }
    public void setTotalCustomers(int totalCustomers) { this.totalCustomers = totalCustomers; }

    public int getTicketReleaseRate() { return ticketReleaseRate; }
    public void setTicketReleaseRate(int ticketReleaseRate) { this.ticketReleaseRate = ticketReleaseRate; }

    public int getCustomerRetrievalRate() { return customerRetrievalRate; }
    public void setCustomerRetrievalRate(int customerRetrievalRate) { this.customerRetrievalRate = customerRetrievalRate; }

    public int getTotalVendors() { return totalVendors; }
    public void setTotalVendors(int totalVendors) { this.totalVendors = totalVendors; }

    @Override
    public String toString() {
        return "Configuration{" +
                "totalTickets=" + totalTickets +
                ", maxTicketCapacity=" + maxTicketCapacity +
                ", totalCustomers=" + totalCustomers +
                ", ticketReleaseRate=" + ticketReleaseRate +
                ", customerRetrievalRate=" + customerRetrievalRate +
                ", totalVendors=" + totalVendors +
                '}';
    }
}